src/solvers/sat/external_sat.cpp

```diff
@@ -46,11 +44,18 @@ void external_satt::write_cnf_file(std::string cnf_file)
   std::ofstream out(cnf_file);
 
   // We start counting at 1, thus there is one variable fewer.
-  out << "p cnf " << (no_variables() - 1) << ' ' << no_clauses() << '\n';
+  out << "p cnf " << (no_variables() - 1) << ' '
+      << no_clauses() + assumptions.size() << '\n';
 
+  // output the problem clauses
   for(auto &c : clauses)
     dimacs_cnft::write_dimacs_clause(c, out, false);
 
+  // output the assumption clauses
+  forall_literals(it, assumptions)
+    if(!it->is_constant())
+      out << it->dimacs() << '\n';
```

Comment:
Isn't this missing a terminating '0' literal on each line, i.e., this should be `<< " 0\n";`